syntax = "proto3";

package test;

message StreamRequest {
  int32 num_messages = 1;
}

message StreamResponse {
  int32 message_num = 1;
  string data = 2;
}

service TestService {
  // Server streaming - sends multiple messages then completes
  rpc StreamData(StreamRequest) returns (stream StreamResponse);
}
